packages/lesswrong/server/repos/CommentsRepo.ts

```diff
@@ -582,22 +586,176 @@ class CommentsRepo extends AbstractRepo<"Comments"> {
+          WHERE ufe."userId" = $(userId)
+            AND ufe."eventType" != 'served'
+            AND ufe."collectionName" = 'Comments'
+            AND ufe."createdAt" > (NOW() - INTERVAL $(lookbackInterval))
+            AND COALESCE(c_views."topLevelCommentId", c_views._id) IN (
+                SELECT "threadTopLevelId_inner_rat" FROM (
+                    SELECT COALESCE(c_inner."topLevelCommentId", c_inner._id) AS "threadTopLevelId_inner_rat", MAX(c_inner."postedAt") AS "lastCommentActivity_inner"
+                    FROM "Comments" c_inner
+                    WHERE c_inner."postedAt" > (NOW() - INTERVAL $(lookbackInterval))
+                      AND c_inner.deleted IS NOT TRUE AND c_inner.retracted IS NOT TRUE AND c_inner."authorIsUnreviewed" IS NOT TRUE
+                    GROUP BY COALESCE(c_inner."topLevelCommentId", c_inner._id)
+                    ORDER BY "lastCommentActivity_inner" DESC
+                    LIMIT $(threadCandidateLimit)
+                ) recent_threads_filter_for_uve
+            )
+          GROUP BY COALESCE(c_views."topLevelCommentId", c_views._id)
+        ) userViewEventsInThreads ON recentActiveThreads."threadTopLevelId" = userViewEventsInThreads."threadTopLevelId"
+      LEFT JOIN
+        ( -- get threads that are on posts read by the user
+          SELECT DISTINCT COALESCE(c_read."topLevelCommentId", c_read._id) AS "threadTopLevelId"
+          FROM "Comments" c_read
+          JOIN (
+            SELECT "postId"
+            FROM "ReadStatuses" rs
+            WHERE rs."userId" = $(userId)
+              AND rs."isRead" IS TRUE
+              AND rs."lastUpdated" > (NOW() - INTERVAL $(lookbackInterval))
+            UNION DISTINCT
+            SELECT DISTINCT "documentId" AS "postId"
+            FROM "UltraFeedEvents" ufe_posts
+            WHERE ufe_posts."userId" = $(userId)
+              AND ufe_posts."collectionName" = 'Posts'
+              AND ufe_posts."eventType" != 'served'
+              AND ufe_posts."createdAt" > (NOW() - INTERVAL $(lookbackInterval))
+          ) "readPosts_subquery" ON c_read."postId" = "readPosts_subquery"."postId"
+          WHERE c_read."postedAt" > (NOW() - INTERVAL $(lookbackInterval))
+            AND COALESCE(c_read."topLevelCommentId", c_read._id) IN (
+                SELECT "threadTopLevelId_inner_rat" FROM (
+                    SELECT COALESCE(c_inner."topLevelCommentId", c_inner._id) AS "threadTopLevelId_inner_rat", MAX(c_inner."postedAt") AS "lastCommentActivity_inner"
+                    FROM "Comments" c_inner
+                    WHERE c_inner."postedAt" > (NOW() - INTERVAL $(lookbackInterval))
+                      AND c_inner.deleted IS NOT TRUE AND c_inner.retracted IS NOT TRUE AND c_inner."authorIsUnreviewed" IS NOT TRUE
+                    GROUP BY COALESCE(c_inner."topLevelCommentId", c_inner._id)
+                    ORDER BY "lastCommentActivity_inner" DESC
+                    LIMIT $(threadCandidateLimit)
```

Comment:
I think the fact that this is repeated in each subquery suggests that something can be simplified, but can happen later